project/ExampleSuiteCompilationProject.scala

```diff
@@ -1,12 +1,58 @@
 import sbt._
 import sbt.Keys._
 import java.io.File
-
+import de.heikoseeberger.sbtheader.HeaderPlugin
+import org.scalafmt.sbt.ScalafmtPlugin
+import sbtprotoc.ProtocPlugin
+import ProtocPlugin.autoImport.PB
+import akka.grpc.sbt.AkkaGrpcPlugin
+import com.lightbend.sbt.JavaFormatterPlugin
 object ExampleSuiteCompilationProject {
-  val empty =
+
+  def compilationProject(lang: String, pathToTests: String)(configureFunc: Project => Project) = {
     new CompositeProject {
-      def componentProjects = Seq.empty[Project]
+
+      lazy val root =
+        Project(id = s"codegen${lang.capitalize}CompilationExampleSuite", base = file(pathToTests))
+          .aggregate(innerProjects.map(p => p: ProjectReference): _*)
+
+      val innerProjects =
+        findProjects
+          .map { case (dir, name) =>
+            Project(s"test-$lang" + name, dir)
+              .disablePlugins(HeaderPlugin, ScalafmtPlugin, JavaFormatterPlugin)
+              .settings(
+                Compile / unmanagedSourceDirectories ++= Seq("generated-managed", "generated-unmanaged").map(
+                  baseDirectory.value / _),
+                Test / unmanagedSourceDirectories ++= Seq("generated-test-managed", "generated-test-unmanaged").map(
+                  baseDirectory.value / _),
+                Compile / PB.protoSources += baseDirectory.value / "proto")
+              .enablePlugins(ProtocPlugin, AkkaGrpcPlugin)
+          }
+          .map(configureFunc) // apply specific settings
```

Comment:
two `.map` to make it more obvious we were applying this function. Anyway, just cosmetics.